### PR TITLE
#149 Implemented validate_required decorator for all models

### PR DESCRIPTION
**Pull request by marceloamadeu**
For PR #149:

The validate_required decorator has been implemented for all models in which require verification of mandatory parameters.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#265](https://github.com/timelesslounge/timelessis/issues/265) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#265](https://github.com/timelesslounge/timelessis/issues/265) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/c9bcae68-c644-4baf-9ab4-d669017632c5); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#265` already assigned to @vryazanov, can't assign to @vryazanov

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#265` already assigned to @vryazanov, can't assign to @vryazanov

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#265` already assigned to @vryazanov, can't assign to @vryazanov

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#265` already assigned to @vryazanov, can't assign to @vryazanov

**Comment by 0crat** [comment]
Job `gh:timelesslounge/timelessis#265` already assigned to @vryazanov, can't assign to @vryazanov

File: timeless/companies/models.py [hunk 1]
```diff
@@ -25,5 +25,7 @@ class Company(TimestampsMixin, DB.Model):
     items = DB.relationship("Item", order_by="Item.id",
                             back_populates="company")
 
+    @validate_required("name", "code")
```
Comment (vryazanov) [inline timeless/companies/models.py]: @marceloamadeu `@` means it's a decorator which decorates a method following in the next line (so plz remove blank line you've added after each decorator :)). So here it decorates `__repr__` and it does not make any sense. This decorator simply checks that arguments of decorated function contain provided to `validate_required` function args

File: timeless/companies/models.py [hunk 2]
```diff
@@ -25,5 +25,12 @@ class Company(TimestampsMixin, DB.Model):
...
+
+    def __init__(self, **kwargs):
```
Comment (vryazanov) [inline timeless/companies/models.py]: @marceloamadeu No need to override attributes with mocked data, just do:
```python
@validate_required("name", "code")
def __init__(self, *args, **kwargs):
    super(Company, self).__init__(**kwargs)
```

By my opinion it looks a bit weird, but this is how `validate_required` implemented.
Please fix it in all places in PR.

**Comment by vryazanov** [comment]
@rultor merge

**Comment by emilianodellacasa** [comment]
@marceloamadeu a test is failing, please check. Also check cod climate, as there is something to fix with quality code

**Comment by emilianodellacasa** [comment]
@marceloamadeu please fix the conflicts

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
Job [#265](https://github.com/timelesslounge/timelessis/issues/265) is not in the agenda of @vryazanov[/z](https://www.0crat.com/u/vryazanov), can't inspect

**Comment by 0crat** [comment]
The job [#265](https://github.com/timelesslounge/timelessis/issues/265) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by vryazanov** [comment]
@0crat status

**Comment by 0crat** [comment]
> @0crat status ([here](https://github.com/timelesslounge/timelessis/issues/265#issuecomment-464996657))

@vryazanov This is what I know about this job in [CF9LG1YCA](https://www.0crat.com/p/CF9LG1YCA), as in [§32](http://www.zerocracy.com/policy.html#32):

  * The job [#265](https://github.com/timelesslounge/timelessis/issues/265) is not in scope
  * The job is waiting quality review verdict by @ypshenychka[/z](https://www.0crat.com/u/ypshenychka) for 16hr
  * @vryazanov[/z](https://www.0crat.com/u/vryazanov) will receive 15 minutes after quality review
  * Job [footprint](https://www.0crat.com/footprint/CF9LG1YCA?q=%7Bjob%3A%27gh%3Atimelesslounge%2Ftimelessis%23265%27%2Ctype%3A%7B%24not%3A%2F%5ENotify%2F%7D%7D) (restricted area)

**Comment by ypshenychka** [comment]
@vryazanov According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only two issues were found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by vryazanov** [comment]
@ypshenychka I confirm

**Comment by ypshenychka** [comment]
@vryazanov thanks

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Pull request merged**